src/overlays/actors/ovl_En_Dg/z_en_dg.c

```diff
@@ -863,41 +1002,54 @@ void func_8098B004(EnDg* this, PlayState* play) {
         Math_ApproachF(&this->actor.speedXZ, 3.5f, 0.2f, 1.0f);
     }
 
-    func_8098A064(this, play);
+    EnDg_CheckForBremenMaskMarch(this, play);
     Actor_MoveWithGravity(&this->actor);
-    func_80989974(this);
-    if (!(this->unk_280 & 0x20)) {
-        func_809899C8(this, 3.0f);
+    EnDg_PlaySfxWalk(this);
+
+    // If we're in this function, the player is either marching with the Bremen Mask, or
+    // they're a Zora. The below code will make the dog repeatedly bark in the latter case.
+    if (!(this->dogFlags & DOG_FLAG_FOLLOWING_BREMEN_MASK)) {
+        EnDg_PlaySfxBark(this, 3.0f);
     }
 }
 
-void func_8098B198(EnDg* this, PlayState* play) {
+/**
+ * Makes the dog slowly back up and face the player before performing a jump attack. Used to
+ * give the dog a bit of time to correct its angle if they were unlikely to hit the player.
+ */
+void EnDg_SlowlyBackUpBeforeAttacking(EnDg* this, PlayState* play) {
     if (this->actor.xzDistToPlayer > 72.0f) {
-        func_80989140(&this->skelAnime, sAnimations, 14);
-        this->actionFunc = func_8098AC34;
+        EnDg_ChangeAnimation(&this->skelAnime, sAnimations, DOG_ANIMATION_JUMP_ATTACK);
+        this->actionFunc = EnDg_JumpAttack;
     }
 
     Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 4, 0x3E8, 1);
     if (this->actor.bgCheckFlags & 8) {
         this->actor.shape.rot.y = this->actor.wallYaw;
-        func_80989140(&this->skelAnime, sAnimations, 14);
-        this->actionFunc = func_8098AC34;
+        EnDg_ChangeAnimation(&this->skelAnime, sAnimations, DOG_ANIMATION_JUMP_ATTACK);
+        this->actionFunc = EnDg_JumpAttack;
     }
 
     this->actor.world.rot.y = this->actor.shape.rot.y;
     Math_ApproachF(&this->actor.speedXZ, -1.0f, 0.2f, 1.0f);
     Actor_MoveWithGravity(&this->actor);
-    func_80989974(this);
-    func_80989A9C(this, 4.0f);
+    EnDg_PlaySfxWalk(this);
+    EnDg_PlaySfxGrowl(this, 4.0f);
 }
 
-void func_8098B28C(EnDg* this, PlayState* play) {
+/**
+ * An unused function that will make the dog back away from the player while growling.
+ * It is quite similar to EnDg_BackAwayFromGoron; perhaps this is an earlier version of it?
+ * Unlike that function, this will make the dog approach the player if they move too far away.
+ * It also lacks the behavior where the dog runs away in a panic after a short time.
+ */
+void EnDg_UnusedBackAwayFromPlayer(EnDg* this, PlayState* play) {
```

Comment:
```suggestion
/**
 * An unused function that will make the dog back away from the player while growling.
 * It is quite similar to EnDg_BackAwayFromGoron; perhaps this is an earlier version of it?
 * Unlike that function, this will make the dog approach the player if they move too far away.
 * It also lacks the behavior where the dog runs away in a panic after a short time.
 */
void EnDg_BackAwayFromPlayer(EnDg* this, PlayState* play) {
```
I think this being unused should be indicated through comments instead of the function name itself? (and you already mention it in the description)